sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StatePartitionReader.scala

```diff
@@ -145,11 +159,24 @@ class StatePartitionReader(
     }
   }
 
+  // TODO we can remove this create new col family after integration with VCF
+  private var stateStore: Option[StateStore] = None
+
   override lazy val iter: Iterator[InternalRow] = {
-    store.iterator().map(pair => unifyStateRowPair((pair.key, pair.value)))
+    if (transformWithState) {
+      stateStore = Option(provider.getStore(partition.sourceOptions.batchId + 1))
+      assert(stateVarName.isDefined, "state variable name has to be defined for TWS")
+      stateStore.get.createColFamilyIfAbsent(
+        stateVarName.get, keySchema, valueSchema, keyStateEncoderType)
+      stateStore.get.iterator(stateVarName.get)
+        .map(pair => unifyStateRowPair((pair.key, pair.value)))
+    } else {
+      store.iterator().map(pair => unifyStateRowPair((pair.key, pair.value)))
+    }
   }
 
   override def close(): Unit = {
+    if (stateStore.isDefined) stateStore.get.abort()
```

Comment:
Why do we need to add this ?

Reply:
Because we opened a statestore [here](https://github.com/apache/spark/pull/47450/files#diff-163dae1387575fe6378b403e5644ef395e294615828d8c93bd30a51e8ae89bceR163 ) and need to call create new column family explicitly. So we'll need to close that state store.